### DISCUSSION TITLE
I could not find any method to delete previous file on FileUpload.

### DISCUSSION THREAD

**Discussion by yuvrajtimalsina** [filamentphp/filament#7541] [Help] [answered]
I've been dealing with file uploads and noticed there's no clear way to delete previously uploaded files using FileUpload. Anyone else faced this? Any workarounds or ideas?

**Comment by pboivin** [comment, marked as answer]
It's something you have to implement yourself, because Filament doesn't know how images are used in the rest of your app. The recommended way is to use a model observer:
https://laravel.com/docs/10.x/eloquent#events

**Reply by yuvrajtimalsina** [reply]
Thank you for the response. I was hoping to get it within the filament itself. Now, since I got to know its not there, I will happily proceed with your suggestion instead.🙇
